data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/ConnectionConfiguration.java

```diff
@@ -270,29 +259,13 @@ public RestHighLevelClient createClient() {
     return new RestHighLevelClient(restClientBuilder);
   }
 
-  private void attachSigV4(final RestClientBuilder restClientBuilder) {
+  private void attachSigV4(final RestClientBuilder restClientBuilder, AwsCredentialsSupplier awsCredentialsSupplier) {
     //if aws signing is enabled we will add AWSRequestSigningApacheInterceptor interceptor,
     //if not follow regular credentials process
     LOG.info("{} is set, will sign requests using AWSRequestSigningApacheInterceptor", AWS_SIGV4);
     final Aws4Signer aws4Signer = Aws4Signer.create();
-    final AwsCredentialsProvider credentialsProvider;
-    if (awsStsRoleArn != null && !awsStsRoleArn.isEmpty()) {
-      AssumeRoleRequest.Builder assumeRoleRequestBuilder = AssumeRoleRequest.builder()
-              .roleSessionName("OpenSearch-Sink-" + UUID.randomUUID())
```

Comment:
@cmanning09 , I do think it will be appropriate to include this information. We can possibly update the documentation as well, but the release notes are the a good place for sure.

Reply:
And yes, we are moving away from the session name per plugin.